app/controllers/submissions_controller.rb

```diff
@@ -87,7 +87,7 @@ def run
       client_socket = tubesock
 
       client_socket.onopen do |_event|
-        kill_client_socket(client_socket) if @embed_options[:disable_run]
+        return kill_client_socket(client_socket) if @embed_options[:disable_run]
```

Comment:
Without returning, the remaining code is still execution. Therefore, these return statements have been added.